airframe-control/src/main/scala/wvlet/airframe/control/Control.scala

```diff
@@ -13,15 +13,38 @@
  */
 package wvlet.airframe.control
 
+import scala.util.control.NonFatal
+
 /**
   *
   */
 object Control {
-  def withResource[Resource <: AutoCloseable, U](resource: Resource)(body: Resource => U): U = {
+  def withResource[R <: AutoCloseable, U](resource: R)(body: R => U): U = {
     try {
       body(resource)
     } finally {
-      resource.close
+      closeQuietly(resource)
+    }
+  }
+
+  def withResources[R1 <: AutoCloseable, R2 <: AutoCloseable, U](resource1: R1, resource2: R2)(
+      body: (R1, R2) => U): U = {
+    try {
+      body(resource1, resource2)
+    } finally {
+      closeQuietly(resource1)
+      closeQuietly(resource2)
```

Comment:
I think we should close `resource2` even if `resource1.close` throw an faital error. How about nesting one more try-catch?

Reply:
like d8b157182281714d4345c811ecae92b40e5ff776?